src/Stratis.Bitcoin/Utilities/AsyncQueue.cs

```diff
@@ -107,6 +107,20 @@ public void Enqueue(T item)
             }
         }
 
+        /// <summary>
+        /// Gets the items count in the queue. For test purposes only.
```

Comment:
For test purposes only but in the production code like this doesn't appear to be the correct strategy

Reply:
Well, originally it didn't say for testing purposes but later we decided that we should specify that this code is too dangerous to use in normal code unless you know exactly what you are doing. Kind of a safeguard.

Reply:
That sounds bad. Logically that says to me, delete it.

Reply:
It doesnt say "For test purposes only." without a comment. 
Would you like me to rename `Count` to `CountForTestPurposesOnly` and get rid of the comment?

Reply:
No, if it is for test purposes only and is dangerous to leave in the prod code as someone might use it, then I want it removed from the prod code as it implies something wrong with the design.

Perhaps I misunderstand what you mean by test purposes. If it is for a unit test to call then use reflection in the unit test if you really must find out this value.

Reply:
I very much agree that this addition to async queue Is ugly, so I do support removing it and writing that in different manner if possible.